### PR TITLE
Tutoring FA21 update

### PR DESCRIPTION
**Pull request by bri25yu**
See https://github.com/compserv/hknweb/pull/324 for history

### CODE REVIEW

**Comment by bri25yu** [comment]
Generating the tutoring algorithm JSON returns empty lists for adjacent slot preferences

<no picture of replication atm>

**Comment by bri25yu** [comment]
Fixing tutoring algorithm JSON returns empty lists for adjacent slots

[prepare-algorithm-input after fix.json.txt](https://github.com/compserv/hknweb/files/7126242/prepare-algorithm-input.after.fix.json.txt)

File: hknweb/tutoring/views.py [hunk 1]
```diff
@@ -192,82 +214,41 @@ def prepare_algorithm_input(request):
...
-        adjacent.append(slot.get_previous_hour_slot().slot_id)
-        adjacent.append(slot.get_after_hour_slot().slot_id)
-    else:
-        # Gap on both directions
-        # Adjacent will be empty
-        pass
-    return adjacent
 
-@permission_required('tutoring.add_slot', login_url='/accounts/login/')
+def get_adjacent_slot_ids(slot):
```
Comment (ochan1) [inline hknweb/tutoring/views.py:232]: ngl, this is genius!

**Comment by ochan1** [comment]
Alright, LGTM!

**Pull request merged**